addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -152,8 +152,18 @@ options.registry.carousel = options.Class.extend({
     onClone: function () {
         var id = 'myCarousel' + new Date().getTime();
         this.$target.attr('id', id);
-        this.$target.find('[data-slide]').attr('href', '#' + id);
-        this.$target.find('[data-slide-to]').attr('data-target', '#' + id);
+        const $dataSlide = this.$target.find('[data-slide]');
+        const $dataSlideTo = this.$target.find('[data-slide-to]');
+        if ($dataSlide.attr('data-target')) {
+            $dataSlide.attr('data-target', '#' + id);
+        } else if ($dataSlide.attr('href')) {
+            $dataSlide.attr('href', '#' + id);
+        }
```

Comment:
Since you are doing this twice, can you put that in a (inline or not) function?

Reply:
done